plugins/modules/ec2_instance.py

```diff
@@ -1456,7 +1462,13 @@ def value_wrapper(v):
 
     if params.get("security_group") or params.get("security_groups"):
         try:
-            value = client.describe_instance_attribute(aws_retry=True, Attribute="groupSet", InstanceId=id_)
+            value = AWSRetry.jittered_backoff(
```

Comment:
I might be wrong, but adding `InvalidInstanceID.NotFound` https://github.com/ansible-collections/amazon.aws/pull/1650/files#diff-a6b28ca5e2f98a5237ad2e88e544b3df3de7bbc95c8dc9088cae572ff92c2f60R2179 should do the job.